libs/kotaemon/kotaemon/indices/vectorindex.py

```diff
@@ -69,6 +70,13 @@ def run(self, text: str | list[str] | Document | list[Document]):
         if self.doc_store:
             print("Adding documents to doc store")
             self.doc_store.add(input_)
+        # save the text content into markdown format
+        if self.cache_dir is not None:
+            file_name, _ = os.path.splitext(input_[0].metadata['file_name'])
+            for i in range(len(input_)):
+                with open(self.cache_dir / f"{file_name}_{i}.md", "w") as f:
```

Comment:
Since `self.cache_dir` can be string and will not work with `/` operator, should use `Path(self.cache_dir)` instead.